cpp/oneapi/dal/graph/test/container_test.cpp

```diff
@@ -0,0 +1,729 @@
+    return lhs == rhs;
+}
+
+template <typename T, typename Allocator>
+bool compare_content(const vector_container<T, Allocator>& lhs,
+                     const vector_container<T, Allocator>& rhs) {
+    bool equal = lhs.size() == rhs.size();
+    for (std::int64_t i = 0; equal && i < lhs.size(); i++) {
+        equal &= compare_content(lhs[i], rhs[i]);
+    }
+    return equal;
+}
+
+template <typename First, typename Second, typename Third>
+bool compare_content(const std::tuple<First, Second, Third>& lhs,
+                     const std::tuple<First, Second, Third>& rhs) {
+    return compare_content(std::get<0>(lhs), std::get<0>(rhs)) &&
+           compare_content(std::get<1>(lhs), std::get<1>(rhs)) &&
+           compare_content(std::get<2>(lhs), std::get<2>(rhs));
+}
+
+template <typename T>
+void fill_content(T& value, std::int64_t n = 0) {
+    value = static_cast<T>(n);
+}
+
+template <typename First, typename Second, typename Third>
+void fill_content(std::tuple<First, Second, Third>& value, std::int64_t n = 0) {
+    fill_content(std::get<0>(value), n);
+    fill_content(std::get<1>(value), n + 1);
+    fill_content(std::get<2>(value), n + 2);
+}
+
+template <typename T, typename Allocator, template <typename, typename> class Container>
+void fill_content(Container<T, Allocator>& value, std::int64_t n = 0) {
+    if (value.empty()) {
+        value.resize(n);
+    }
+    for (std::int64_t i = 0; i < dal::detail::integral_cast<std::int64_t>(value.size()); i++) {
+        fill_content(value[i], i);
+    }
+}
+
+template <typename T>
+void check_construct_empty() {
+    vector_container<T> vec;
+    REQUIRE(vec.empty());
+    REQUIRE(vec.get_count() == 0);
+    REQUIRE(vec.size() == 0);
+    REQUIRE(vec.capacity() == 1);
+}
+
+template <typename T>
+void check_construct_empty_with_custom_allocator() {
+    allocated_bytes_count = 0;
+    {
+        CountingAllocator<T> alloc;
+        vector_container<T, CountingAllocator<T>> vec(alloc);
+        REQUIRE(vec.empty());
+        REQUIRE(vec.get_count() == 0);
+        REQUIRE(vec.size() == 0);
+        REQUIRE(vec.capacity() == 1);
+    }
+    // REQUIRE(allocated_bytes_count == 0);
+}
+
+template <typename T>
+void check_construct_given_size(std::int64_t n) {
+    vector_container<T> vec(n);
+    REQUIRE(!vec.empty());
+    REQUIRE(vec.get_count() == n);
+    REQUIRE(vec.size() == n);
+    REQUIRE(vec.capacity() == lower_bound_pow(n));
+}
+
+template <typename T>
+void check_construct_given_size_with_custom_allocator(std::int64_t n) {
+    allocated_bytes_count = 0;
+    {
+        CountingAllocator<T> alloc;
+        vector_container<T, CountingAllocator<T>> vec(n, alloc);
+        REQUIRE(!vec.empty());
+        REQUIRE(vec.get_count() == n);
+        REQUIRE(vec.size() == n);
+        REQUIRE(vec.capacity() == lower_bound_pow(n));
+    }
+    // TODO: Uncomment after the memory leak fix
+    // REQUIRE(allocated_bytes_count == 0);
```

Comment:
It seems strange to me that memory leak can affect number of allocated bytes in this test.